Add Watcher APIs for updating/retrieving settings

dakrone opened the pull request:
The `.watches` index is a system index, which means that its settings cannot be modified by the user. This commit adds APIs (`PUT /_watcher/settings` and `GET /_watcher/settings`) that allow modifying and retrieving a subset of index settings for the `.watches` index.

The settings that are currently allowed are `index.number_of_replicas` and `index.auto_expand_replicas`, though more may be added in the future.

Resolves https://github.com/elastic/elasticsearch/issues/92991

dakrone committed:
Add Watcher APIs for updating/retrieving settings

The `.watches` index is a system index, which means that its settings cannot be modified by the
user. This commit adds APIs (`PUT /_watcher/settings` and `GET /_watcher/settings`) that allow
modifying and retrieving a subset of index settings for the `.watches` index.

The settings that are currently allowed are `index.number_of_replicas` and
`index.auto_expand_replicas`, though more may be added in the future.

Resolves #92991

elasticsearchmachine commented:
Pinging @elastic/es-data-management (Team:Data Management)

dakrone committed:
Update docs/changelog/95342.yaml

elasticsearchmachine commented:
Hi @dakrone, I've created a changelog YAML for you.

dakrone committed:
Update API as non-operator

dakrone committed:
Add a watch so .watches is created in docs tests

gmarouli commented on x-pack/docs/en/rest-api/watcher/get-settings.asciidoc:
```diff
@@ -0,0 +1,20 @@
+
+This API allows a user to retrieve the user-configurable settings for the Watcher internal index (`.watches`). Only a subset of the index settings—those that are use-configurable—will be shown. This includes:
+
+- `index.auto_expand_replias`
```
```suggestion
- `index.auto_expand_replicas`
```

gmarouli commented on x-pack/docs/en/rest-api/watcher/get-settings.asciidoc:
```diff
@@ -0,0 +1,20 @@
+- `index.auto_expand_replias`
+- `index.number_of_replicas`
+
+And example of retrieving the Watcher settings:
```
I am not sure about this one, but I think an example works better:
```suggestion
An example of retrieving the Watcher settings:
```

gmarouli commented on x-pack/docs/en/rest-api/watcher/update-settings.asciidoc:
```diff
@@ -0,0 +1,49 @@
+
+This API allows a user to modify the settings for the Watcher internal index (`.watches`). Only a subset of settings are allowed to by modified. This includes:
+
+- `index.auto_expand_replias`
```
```suggestion
- `index.auto_expand_replicas`
```

gmarouli commented on x-pack/docs/en/rest-api/watcher/update-settings.asciidoc:
```diff
@@ -0,0 +1,49 @@
+- `index.auto_expand_replias`
+- `index.number_of_replicas`
+
+And example of modifying the Watcher settings:
```
Probably here too "an" instead of "and"?

gmarouli commented on x-pack/docs/en/rest-api/watcher/get-settings.asciidoc:
```diff
@@ -0,0 +1,20 @@
+<titleabbrev>Get Watcher settings</titleabbrev>
+++++
+
+This API allows a user to retrieve the user-configurable settings for the Watcher internal index (`.watches`). Only a subset of the index settings—those that are use-configurable—will be shown. This includes:
```
Did you mean user-configurable here instead of use-configurable?

gmarouli reviewed (commented):
Thanks for working on this Lee, so happy to give this to our customers who could not scale watcher anymore. 

It looks good in general, I only have some comments about some typos in the docs and one addition: I think we should list the two new APIs in `x-pack/docs/en/rest-api/watcher.asciidoc`. What do you think?

dakrone committed:
Add settings APIs to Watcher top level overview docs

dakrone committed:
Fix doc typos

dakrone commented:
> I think we should list the two new APIs in x-pack/docs/en/rest-api/watcher.asciidoc. What do you think?

I totally forgot about that, thanks for the reminder! I added it there and fixed the typos; thanks for taking a look!

gmarouli commented:
> I totally forgot about that, thanks for the reminder! I added it there and fixed the typos; thanks for taking a look!

I have been working on the docs for DLM, so that helped ;). I am going to take one more "sanity" look at the doc preview but after that you are good to go.

dakrone committed:
Fix anchor in get-settings.asciidoc

gmarouli reviewed (approved):
LGTM 🚀  thanks again @dakrone!
